lib/Widgets/MessageDialog.vala

```diff
@@ -0,0 +1,240 @@
+ *      "This is a secondary, multiline, long text. This text usually extends the primary text and prints e.g: the details of an error.",
+ *      "applications-development",
+ *      Gtk.ButtonsType.CLOSE
+ *   );
+ *
+ *   var custom_widget = new Gtk.CheckButton.with_label ("Custom widget");
+ *   custom_widget.show ();
+ *
+ *   message_dialog.message_grid.attach (custom_widget, 1, 2, 1, 1);
+ *   message_dialog.run ();
+ *   message_dialog.destroy ();
+ * }}}
+ * 
+ * {{../../doc/images/MessageDialog.png}}
+ */
+public class Granite.MessageDialog : Gtk.Dialog {
+    /**
+     * The primary text, title of the dialog.
+     */
+    public string primary_text { 
+        get {
+            return primary_label.label;
+        }
+
+        set {
+            primary_label.label = value;
+        }
+    }
+
+    /**
+     * The secondary text, body of the dialog.
+     */
+    public string secondary_text { 
+        get {
+            return secondary_label.label;
+        }
+
+        set {
+            secondary_label.label = value; 
+        }
+    }
+
+    /**
+     * The {@link GLib.Icon} that is used to display the image
+     * on the left side of the dialog.
+     */
+    public GLib.Icon image_icon { 
+        owned get {
+            return image.gicon;
+        }
+        
+        set {
+            image.set_from_gicon (value, Gtk.IconSize.DIALOG);
+        }
+    }
+
+    /**
+     * The {@link Gtk.Label} that displays the {@link Granite.MessageDialog.primary_text}.
+     * 
+     * Most of the times, you will only want to modify the {@link Granite.MessageDialog.primary_text} string, 
+     * this is available to set additional properites like {@link Gtk.Label.use_markup} if you wish to do so.
+     */
+    public Gtk.Label primary_label { get; construct; }
+
+    /**
+     * The {@link Gtk.Label} that displays the {@link Granite.MessageDialog.secondary_text}.
+     *
+     * Most of the times, you will only want to modify the {@link Granite.MessageDialog.secondary_text} string, 
+     * this is available to set additional properites like {@link Gtk.Label.use_markup} if you wish to do so.
+     */
+    public Gtk.Label secondary_label { get; construct; }
+
+    /**
+     * The {@link Gtk.ButtonsType} value to display a set of buttons
+     * in the dialog.
+     * 
+     * By design, some actions are not acceptable and such action values will not be added to the dialog, these include:
+     *
+     *  * {@link Gtk.ButtonsType.OK}
+     *  * {@link Gtk.ButtonsType.YES_NO}
+     *  * {@link Gtk.ButtonsType.OK_CANCEL}
+     *
+     * If you wish to provide more specific actions for your dialog
+     * pass a {@link Gtk.ButtonsType.NONE} to {@link Granite.MessageDialog} and manually
+     * add those actions with {@link Gtk.Dialog.add_buttons} or {@link Gtk.Dialog.add_action_widget}.
+     */
+    public Gtk.ButtonsType buttons { 
+        construct {
+            switch (value) {
+                case Gtk.ButtonsType.NONE:
```

Comment:
Thinking about it, `NONE` probably isn't a sane option because there would be no way to close the window

Reply:
`NONE` has to be supported because, suppose you want to add your own buttons instead of the built-in ones.

Reply:
Oops you're right